### DISCUSSION TITLE
Is it possible to have `.*?`-like fallback behavior in `alt`?

### DISCUSSION THREAD

**Discussion by zharinov** [winnow-rs/winnow#250] [Q&A]
For example, if I want to use `alt(("foo", "bar", <else>))` for the `foobar...baz...foobar` input, the output would be: `foo`, `bar`, `...baz...`, `foo` and `bar`.

**Comment by zharinov** [comment]
Actually, seems like it can be quite trivially implemented on the user side, by slicing the input char-by-char from the beginning until the `alt(("foo", "bar"))` matches something.

**Comment by epage** [comment]
Without a more complete code sample, it isn't fully clear what you are trying to accomplish though https://docs.rs/winnow/latest/winnow/combinator/fn.repeat_till0.html might be useful.

**Reply by zharinov** [reply]
I want to create tokenizer which produces `Token::UnknownFragment(&str)` when it encounters symbols that weren't described, i.e. for doing sort of incomplete parsing. Seems like fn you've pointed to could do the trick 👍
